### PR TITLE
MB-61719: Windows requires index variable in an OpenMP for to be signed

### PR DESCRIPTION
**Pull request by abhinavdangeti**
See: https://learn.microsoft.com/en-us/cpp/error-messages/compiler-errors-2/compiler-error-c3016?view=msvc-170

### CODE REVIEW

**Comment by abhinavdangeti** [comment]
It seems we can get around this issue by using clang instead of msvc.

File: faiss/utils/distances_fused/simdlib_based.cpp [hunk 1]
```diff
@@ -257,17 +257,19 @@ void exhaustive_L2sqr_fused_cmax(
...
     }
 
-    const size_t nx_p = (nx / NX_POINTS_PER_LOOP) * NX_POINTS_PER_LOOP;
+    const ssize_t nx_p = (ssize_t)((nx / NX_POINTS_PER_LOOP) * NX_POINTS_PER_LOOP);
     // the main loop.
 #pragma omp parallel for schedule(dynamic) num_threads(num_omp_threads)
-    for (size_t i = 0; i < nx_p; i += NX_POINTS_PER_LOOP) {
+    // use signed instead of size_t because of C3016 error on windows:
+    // https://learn.microsoft.com/en-us/cpp/error-messages/compiler-errors-2/compiler-error-c3016
+    for (ssize_t i = 0; i < nx_p; i += (ssize_t)NX_POINTS_PER_LOOP) {
```
Comment (CascadingRadium) [inline faiss/utils/distances_fused/simdlib_based.cpp:265]: this works if `ssize_t` is defined and available (it is non standard)
pls refer to facebook/faiss commit https://github.com/facebookresearch/faiss/pull/2889
that fixed the same issue in another file

File: faiss/utils/distances_fused/simdlib_based.cpp [hunk 2]
```diff
@@ -257,17 +257,19 @@ void exhaustive_L2sqr_fused_cmax(
...
         kernel<DIM, NX_POINTS_PER_LOOP, NY_POINTS_PER_LOOP>(
-                x, y, y_transposed.data(), ny, res, y_norms, i);
+                x, y, y_transposed.data(), ny, res, y_norms, (size_t)i);
     }
 
-    for (size_t i = nx_p; i < nx; i++) {
+    for (ssize_t i = nx_p; i < (ssize_t)nx; i++) {
```
Comment (CascadingRadium) [inline faiss/utils/distances_fused/simdlib_based.cpp:270]: it is not required to add any changes to this line as
`pragma omp for` only applies to the immediate next for loop (which is the one above)
this for loop is not under openmp

**Comment by abhinavdangeti** [comment]
We won't be needing this anymore, as we're pivoting to using clang to compile windows builds - which fixes the compilation error called out.

**Pull request closed without merging**